modules/service/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql

```diff
@@ -3607,64 +3565,168 @@ enum GmosNorthGrating {
+  ampGain: GmosAmpGain!
+
+  # Default GMOS amp gain (LOW).
+  defaultAmpGain: GmosAmpGain!
+
+  # Optional explicitly specified GMOS amp gain.  If set it override the default.
   explicitAmpGain: GmosAmpGain
 
-  # Explicitly specified GMOS ROI, overriding the default (FULL_FRAME)
+
+  # GMOS ROI, either explicitly specified in explicitRoi or else taken from the
+  # defaultRoi.
+  roi: GmosRoi!
+
+  # Default GMOS ROI (FULL_FRAME).
+  defaultRoi: GmosRoi!
+
+  # Optional explicitly specified GMOS ROI. If set it overrides the default.
   explicitRoi: GmosRoi
 
-  # Explicitly specified wavelength dithers required to fill in the chip gaps (in nm), taking the place of the calculated value based on the grating dispersion
+
+  # Wavelength dithers required to fill in the chip gaps, expressed in nanometers.
+  # This value is either explicitly specified in explicitWavelengthDithersNm
+  wavelengthDithersNm: [BigDecimal!]!
+
+  # Default wavelength dithers, calculated based on the grating dispersion.
+  defaultWavelengthDithersNm: [BigDecimal!]!
+
+  # Optional explicitly specified wavelength dithers.  If set it overrides the
+  # default.
   explicitWavelengthDithersNm: [BigDecimal!]
 
-  # Explicitly specified spatial q offsets, overriding the default
+
+  # Spacial q offsets, either explicitly specified in explicitSpatialOffsets
+  # or else taken from defaultSpatialOffsets
+  spatialOffsets: [q!]!
+
+  # Default spatial offsets.
+  defaultSpatialOffsets: [q!]!
+
+  # Optional explicitly specified spatial q offsets. If set it overrides the
+  # the default.
   explicitSpatialOffsets: [q!]
+
 }
 
-# GMOS North Long Slit basic configuration
-type GmosNorthLongSlitBasic {
+# GMOS North Long Slit mode
+type GmosNorthLongSlit implements GmosLongSlitObservingMode {
+
   # GMOS North Grating
   grating: GmosNorthGrating!
 
   # GMOS North Filter
   filter: GmosNorthFilter
 
   # GMOS North FPU
-  fpu: GmosNorthBuiltinFpu!
+  fpu: GmosNorthBuiltinFpu
+
+  overrideExposureTimeMode: ExposureTimeMode
+
+  centralWavelength: Wavelength!
+  defaultCentralWavelength: Wavelength!
+  explicitCentralWavelength: Wavelength
+
+
+  xBin: GmosXBinning!
+  defaultXBin: GmosXBinning!
+  explicitXBin: GmosXBinning
+
+  yBin: GmosYBinning!
+  defaultYBin: GmosYBinning!
+  explicitYBin: GmosYBinning
+
+  ampReadMode: GmosAmpReadMode!
+  defaultAmpReadMode: GmosAmpReadMode!
+  explicitAmpReadMode: GmosAmpReadMode
+
+  ampGain: GmosAmpGain!
+  defaultAmpGain: GmosAmpGain!
+  explicitAmpGain: GmosAmpGain
+
+  roi: GmosRoi!
+  defaultRoi: GmosRoi!
+  explicitRoi: GmosRoi
+
+  wavelengthDithersNm: [BigDecimal!]!
+  defaultWavelengthDithersNm: [BigDecimal!]!
+  explicitWavelengthDithersNm: [BigDecimal!]
+
+  spatialOffsets: [q!]!
+  defaultSpatialOffsets: [q!]!
+  explicitSpatialOffsets: [q!]
+
+  # The grating as it was initially selected.  See the `grating` field for the
+  # grating that will be used in the observation.
+  initialGrating: GmosNorthGrating!
```

Comment:
Andy expressed a desire to go back to the table of options in the UI and have the value you originally selected (before any edits via the `GmosNorthLongslitInput`) highlighted.  Perhaps it is possible to do this purely in the UI if it is limited to a single session?  Otherwise we need to keep up with the initial grating, filter and FPU.  Note that these are read-only so there's never any confusion about how to change the grating, filter, or FPU.